### PR TITLE
KAFKA-3069: Fix recursion in ZkSecurityMigrator

### PR DESCRIPTION
**Pull request by fpj**
I'm also fixing a bug in the testChroot test case.

### CODE REVIEW

File: core/src/test/scala/unit/kafka/security/auth/ZkAuthorizationTest.scala [hunk 1]
```diff
@@ -197,7 +198,7 @@ class ZkAuthorizationTest extends ZooKeeperTestHarness with Logging{
       case true => "secure"
       case false => "unsecure"
     }
-    ZkSecurityMigrator.run(Array(s"--zookeeper.acl=$secureOpt", s"--zookeeper.connect=$zkConnect"))
+    ZkSecurityMigrator.run(Array(s"--zookeeper.acl=$secureOpt", s"--zookeeper.connect=$zkUrl"))
     info("Done with migration")
     for (path <- secondZk.securePersistentZkPaths) {
```
Comment (ewencp) [inline core/src/test/scala/unit/kafka/security/auth/ZkAuthorizationTest.scala:211]: Seems like the test hasn't really changed aside from adjusting the `zookeeper.connect` URL. It looks like we're validating the ACLs on paths in `securePersistentZkPaths`, but given this change, shouldn't we also validate that we didn't modify ACLs on other paths?

**Comment by ewencp** [comment]
@fpj Looks simple enough, just one question about tests to validate the new behavior.

File: core/src/test/scala/unit/kafka/security/auth/ZkAuthorizationTest.scala (hunk 1 above)
Comment (fpj) [inline core/src/test/scala/unit/kafka/security/auth/ZkAuthorizationTest.scala:211]: @ewencp good suggestion, I've changed accordingly. let me know if it looks good to you.

**Comment by ewencp** [comment]
Great, LGTM. Thanks @fpj!

**Pull request closed without merging**